src/splitter.ts

```diff
@@ -302,6 +303,7 @@ export const initializeSplitting = async (
       originMessageId.toString(),
       sendCallback
     );
+    acknowledgeMessage();
```

Comment:
Hmm... What if you collect all the promises from send() calls to one array. Once you know the array has been filled, so after potentially sending servicing and non-servicing vehicle messages, wait on all of the promises. Once they have all been fulfilled, check their state. If all of them are successful, call acknowledgeMessage(). If any of the send() calls failed, do not acknowledge and instead throw an exception and let it kill the service because it's either a programming error that we need to catch early or some weird cluster or networking problem.

Also, do not let the waiting on that array of promises stop us from taking the next gtfsrt-vp message in, once our local cache is up-to-date. The Pulsar client should make sure that the send() calls for the first full gtfsrt-vp message and the corresponding ack resolve before the send() calls and the ack for the next gtfsrt-vp message.